provisioner/terraform/modules.go

```diff
@@ -103,6 +103,13 @@ func GetModulesArchive(root fs.FS) ([]byte, error) {
 			if !fileMode.IsRegular() && !fileMode.IsDir() {
 				return nil
 			}
+
+			if fileMode.IsDir() && d.Name() == ".git" {
+				// .git directories are not needed in the archive and only cause
+				// hash differences for identical modules.
```

Comment:
I usually prefer to put these kinds of comments outside of the `if`. Then the explanation actually precedes the condition, rather than explaining it after the fact.